src/Microsoft.TestPlatform.Common/Utilities/ExceptionUtilities.cs

```diff
@@ -22,15 +23,16 @@ public static string GetExceptionMessage(Exception exception)
                 return string.Empty;
             }
 
-            var exceptionString = exception.Message;
+            var exceptionString = new StringBuilder(exception.Message);
```

Comment:
You are right, mostly one or two inner exceptions. But in my opinion it is good pattern to use StringBuilder in all cases when we concatenate the strings.

Reply:
https://blog.codinghorror.com/the-sad-tragedy-of-micro-optimization-theater/ :) I don't think there is any hard rule for that. Usually 10 strings is the threshold. But if you are okay with trading a bit of perf for smaller memory foot print then why not. Not that it would matter in this case because the impact will be negligible (both ways).